qutebrowser/mainwindow/statusbar/zoom.py

```diff
@@ -0,0 +1,28 @@
+"""Zoom percentage displayed in the statusbar."""
+
+from qutebrowser.browser import browsertab
+from qutebrowser.mainwindow.statusbar import textbase
+from qutebrowser.qt.core import pyqtSlot, QObject
+from qutebrowser.config import config
+
+
+class Zoom(textbase.TextBase):
+
+    """Shows zoom percentage in current tab."""
+
+    def __init__(self, parent: QObject = None) -> None:
+        super().__init__(parent)
+        self.on_zoom_changed(1)
+
+    @pyqtSlot(float)
+    def on_zoom_changed(self, factor: float) -> None:
+        """Update percentage when factor changed."""
+        if factor == 1 and config.val.statusbar.zoom.show == 'non-default':
+            self.setText("")
```

Comment:
This causes a gap (because of the padding of the widget) which is a really minor thing, but at the same thing I can't unsee it after I noticed (less noticeable at bigger font sizes):

<img width="343" height="66" alt="Image" src="https://github.com/user-attachments/assets/9d24c104-c85a-4bc6-a7e2-d3176e592d38" />

If it's easy to fix I'd prefer to do that. I think it might just be a matter of calling `self.hide()` here, `self.show()` below in addition to `.setText()`, and adjusting the tests accordingly?